boards/arm/xiao_ble/board.cmake

```diff
@@ -0,0 +1,9 @@
+# SPDX-License-Identifier: Apache-2.0
+
+board_runner_args(nrfjprog "--nrf-family=NRF52")
```

Comment:
I've seen `--softreset` added to the `nrfjprog` args in other nrf boards.  Without it, it doesn't seem to reset the board as I would expect.  Maybe that's not the behavior as others would expect though?

since it doesn't seem to restart the board as I would expect.

Reply:
`--softreset` is required only if you don't have hardware reset connected. I think that when I have RST (pogo pin) connected to J-Link, then XIAO BLE was reset automatically after flashing.

Reply:
ah, you're right. the expansion board doesn't expose reset in the header for some reason.  it's got TX, RX, SWDIO, SWCLK, 3v3,5v, and 2 GND, but NOT reset. The reset is exposed to a button instead.

I guess we can just leave it the way it is until there's something explicit about the expansion board. That'd be for a separate PR.